elixir/apps/api/lib/api/device/channel.ex

```diff
@@ -80,15 +82,23 @@ defmodule API.Device.Channel do
   end
 
   @impl true
-  def handle_in("list_relays", %{"resource_id" => resource_id}, socket) do
+  def handle_in("prepare_connection", %{"resource_id" => resource_id} = attrs, socket) do
+    connected_gateway_ids = Map.get(attrs, "connected_gateway_ids", [])
+
     with {:ok, resource} <- Resources.fetch_resource_by_id(resource_id, socket.assigns.subject),
          # :ok = Resource.authorize(resource, socket.assigns.subject),
+         {:ok, [_ | _] = gateways} <-
+           Gateways.list_connected_gateways_for_resource(resource),
          {:ok, [_ | _] = relays} <- Relays.list_connected_relays_for_resource(resource) do
+      gateway = Gateways.load_balance_gateways(gateways, connected_gateway_ids)
+
       reply =
         {:ok,
          %{
            relays: Views.Relay.render_many(relays, socket.assigns.subject.expires_at),
-           resource_id: resource_id
+           resource_id: resource_id,
+           gateway_id: gateway.id,
+           gateway_remote_ip: gateway.last_seen_remote_ip
```

Comment:
@conectado I've also added a remote IP so that you can add it as an extra ICE candidate once you have time to take care of it.